src/rpg_database/character_sheet/FocusesSet.java

```diff
@@ -0,0 +1,171 @@
+package rpg_database.character_sheet;
+
+import java.util.ArrayList;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import rpg_database.character_sheet.exceptions.InvalidFocusesSetException;
+import rpg_database.character_sheet.interfaces.CustomSetter;
+
+public class FocusesSet implements Set<Focus>, CustomSetter<FocusesSet> {
+
+	private HashSet<Focus> focusesSet;
+
+	public FocusesSet() {
+		focusesSet = new HashSet<Focus>();
+	}
+
+	public FocusesSet(Focuses... focuses) {
+		focusesSet = new HashSet<Focus>();
+		focusesSet = findDuplicates(focuses);
+
+	}
+
+	@Override
+	public Class<FocusesSet> getImplementingClass() {
+		return FocusesSet.class;
+	}
+
+	@Override
+	public void setSelfInSheet(CharacterSheet characterSheet) {
+		characterSheet.characterData.put(Fields.FOCUSES, this);
+	}
+
+	@Override
+	public int size() {
+		return focusesSet.size();
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return focusesSet.isEmpty();
+	}
+
+	@Override
+	public boolean contains(Object o) {
+		return focusesSet.contains(o);
+	}
+
+	@Override
+	public Iterator<Focus> iterator() {
+		return focusesSet.iterator();
+	}
+
+	@Override
+	public Object[] toArray() {
+		return focusesSet.toArray();
+	}
+
+	@Override
+	public <T> T[] toArray(T[] a) {
+		return focusesSet.toArray(a);
+	}
+
+	/**
+	 * Does some thing in old style.
+	 *
+	 * @deprecated use {@link #new()} instead.
+	 */
+	@Deprecated
+	@Override
+	public boolean add(Focus focus) {
+		return focusesSet.add(focus);
+	}
+
+	public boolean add(Focuses focus) {
+		Focus focusInDb = getFocus(focus, focusesSet);
+		if (focusInDb == null) {
+			Focus tempFocus = new Focus(focus);
+			focusesSet.add(tempFocus);
+			return tempFocus.isFocusImproved();

```

Comment:
Itt az add false-ot fog visszaadni, akkor is ha újat adsz hozzá. Miért?

Reply:
Átírva, hülyéseget implementáltam, nem vettem figyelembe az add eredeti return funkciónalitását.